File: TwoSum/TwoSum.py
class Solution:
    # @return a tuple, (index1, index2)
    def twoSum(self, num, target):
        d = {}
        index1 = 0
        index2 = 0
        for index, value in enumerate(num):
            print index, value
            if target - value in d:
                print "get result"
                index1 = d[target - value]
                index2 = index
                break
            else:
                print "not get result", value
                d[value] = index
            print d
        return (index1, index2)

if __name__ == "__main__":
    print Solution().twoSum([3, 2, 4], 6)

File: TwoSum/TwoSum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int> &numbers, int target) {
        vector<int> result;
        map<int, int> m;
        for(int i = 0; i<numbers.size(); i++){
            if(m.find(numbers[i]) == m.end()){
                m[target - numbers[i]] = i;
            }else{
                result.push_back(m[numbers[i]]+1);
                result.push_back(i+1);
            }
        }
        return result;
    }
};


class Solution {
public:
    vector<int> twoSum(vector<int> &numbers, int target) {
        vector<int> result;
        map<int, int> m;
        for(int i = 0; i<numbers.size(); i++){
            if(m.find(target - numbers[i]) == m.end()){
                m[numbers[i]] = i;
            }else{
                result.push_back(m[numbers[i]]+1);
                result.push_back(i+1);
            }
        }
        return result;
    }
};